### PR TITLE
Intune vpn configuration policy android

### PR DESCRIPTION
**Pull request by dannyKBjj**
#### Pull Request (PR) description
Adds support for microsoft.graph.androidDeviceOwnerVpnConfiguration and microsoft.graph.androidVpnConfiguration resource type.

#### This Pull Request (PR) fixes the following issues
None

Note: resource type and actual display in Intune GUI is reversed - i.e. microsoft.graph.androidDeviceOwnerVpnConfiguration handles Enterprise policies. Have named resources in line with resource type.

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [x] Resource parameter descriptions added/updated in the schema.mof.
- [x] Resource documentation added/updated in README.md.
- [x] Resource settings.json file contains all required permissions.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated.
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

### CODE REVIEW

**Comment by dannyKBjj** [comment]
Unit tests say 

Error: [-] Should compile example 'IntuneVPNConfigurationPolicyAndroidDeviceOwner\1-Create.ps1' correctly 696ms (695ms|1ms)
Message
Error: [-] Should compile example 'IntuneVPNConfigurationPolicyAndroidDeviceOwner\2-Update.ps1' correctly 731ms (731ms|0ms)

However, I'm able to compile the examples just fine...

**Comment by ricmestre** [comment]
@dannyKBjj Welcome to the DSC world! The problem is that the CIM instance *MSFT_targetedMobileApps* is already defined somewhere else so if you reuse it with exactly the same name it must follow the same number of properties and they must also have the same names/types.

In this case it was actually you who defined this CIM instance in *MSFT_IntuneVPNConfigurationPolicyIOS* and instead of *name* you called it *address*, most likely not your intention. Assuming that you didn't want to do that just go and replace *address* by *name* in *MSFT_IntuneVPNConfigurationPolicyIOS* and the examples included in this PR should work.

image

**Comment by dannyKBjj** [comment]
Ah!! See https://github.com/microsoft/Microsoft365DSC/pull/5509 

I realised whilst doing my Android modules that I'd messed up defining the targetMobileApps in my iOS VPN module. I've got another pull request that'll bring them into line with each other. I don't actually think targetedMobileApps is used for iOS VPN despite being documented as belonging to the resource class (at least I can't see where it could ever be set) and therefore the issue never came up in testing. It did however come up when I tested my Android VPN configurations, so I fixed it in both. 'name' is the correct value:

see: https://learn.microsoft.com/en-us/graph/api/resources/intune-deviceconfig-applistitem?view=graph-rest-beta 

I guess I should leave this until pull request 5509 is approved and then this issue will fix itself?

**Comment by ricmestre** [comment]
Yes, just went through the other PR, if that one gets merged this will fix itself once you merge the Dev branch into it with those updates but you'll have to rely on the fact that the other PR needs to be merged first.

**Comment by ykuijs** [comment]
Other PR is merged. Please update with Dev to trigger the unit tests again

**Comment by dannyKBjj** [comment]
thanks, seems to have passed the unit tests this time.

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,14 @@
   * Fixed conditional logic for creation and update.
 * IntuneTrustedRootCertificateIOS
   * Initial release
+* IntuneVPNConfigurationPolicyAndroidDeviceOwner
+  * Initial release
+* IntuneVPNConfigurationPolicyAndroidEnterprise
+  * Initial release
+* M365DSCDRGUtil
```
Comment (ricmestre) [inline CHANGELOG.md]: This was part of a PR from @FabienTschanz which was already merged, please remove these 2 entries from the changelog

Comment (dannyKBjj) [reply CHANGELOG.md]: removed 

* IntuneVPNConfigurationPolicyAndroidEnterprise
  * Initial release

Comment (ricmestre) [reply CHANGELOG.md]: No, that's wrong. Add back your entry, what you need to remove is M365DSCDRGUtil/M365DSCUtil entries

Comment (dannyKBjj) [reply CHANGELOG.md]: Ok, not sure how they ended up as part of my commit, but I've removed them! Hopefully done the right thing this time?

Comment (ricmestre) [reply CHANGELOG.md]: yeah looks ok now :)

**Pull request merged**